rtengine/diagonalcurves.cc

```diff
@@ -454,7 +447,13 @@ double DiagonalCurve::getVal (double t) const
             // add highlights effect:
             double htv = xexp(max(mhc * xlog((stretched - mfc) / (1.0 - mfc)),-236.0));
             double hbase = pfull_alt (htv, 0.5, x[4]);
-            return mfc + (1.0 - mfc) * xexp(xlog(hbase) / mhc);
+            //this part of the curve isn't affected by highlight, return the base curve
+            if (hbase < 1e-14 ){ 
```

Comment:
This is a great idea and it works very well. In my tests you could even set a much higher threshold, even something like `1e-5` gives virtually negligible differences and a perfectly smooth curve. The benefit is a small speedup because you don't have to calculate `hbase`.